### PR TITLE
RHEL 8: align with CIS 3, section 2

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

- adjust the section 2 of the CIS policy file to be aligned with version 3.0.0
- update rule references
- update the PCI-DSS profile stability test because:
    -  a rule was selected in PCI-DSS, but it was missing the prodtype for RHEL 8
    - the prodtype got added in this PR which is related to CIS
    - therefore, the rule got added into the PCI-DSS profile as well and the test has to be updated

#### Rationale:

- effort to keep CIS profiles up-to-date

#### Review Hints:

- download the CIS for RHEL 8 version 3
- verify section 2 is aligned

### CODE REVIEW

File: controls/cis_rhel8.yml [hunk 1]
```diff
@@ -774,26 +774,25 @@ controls:
...
     status: automated
     rules:
-      - package_cups_removed
+      - package_dhcp_removed
     related_rules:
-      - service_cups_disabled
-
-  - id: 2.2.5
-    title: Ensure DHCP Server is not installed (Automated)
+      - service_dhcpd_disabled
```
Comment (marcusburghardt) [inline controls/cis_rhel8.yml:786]: Could you also update the reference in "service_dhcpd_disabled", please? It shows 2.2.5.

File: controls/cis_rhel8.yml [hunk 2]
```diff
@@ -971,15 +971,14 @@ controls:
...
   - id: 2.2.20
-    title:  Ensure rsync is not installed or the rsyncd service is masked (Automated)
+    title: Ensure X window server services are not in use (Automated)
     levels:
-      - l1_server
-      - l1_workstation
+      - l2_server
     status: automated
+    notes: >-
+      The rule also configures correct run level to prevent unbootable system.
```
Comment (marcusburghardt) [inline controls/cis_rhel8.yml:979]: The runlevel is configured by `xwindows_runlevel_target`, but it is not selected here.

File: controls/cis_rhel8.yml [hunk 3]
```diff
@@ -971,15 +971,14 @@ controls:
...
     rules:
-      - package_rsync_removed
-    related_rules:
-      - service_rsyncd_disabled
+      - xwindows_remove_packages
```
Comment (marcusburghardt) [inline controls/cis_rhel8.yml:981]: This rule removes 3 packages but the policy asks to remove only `xorg-x11-server-common`.
Wouldn't be better to select `package_xorg-x11-server-common_removed` here?

**Review by marcusburghardt — CHANGES_REQUESTED** [review]
Thanks for the updates. There are only very few comments to be checked. And the conflicts should be relatively easy to be resolved after rebasing.

File: controls/cis_rhel8.yml [hunk 4]
```diff
@@ -754,14 +754,16 @@ controls:
...
 
   - id: 2.2.1
-    title: Ensure xinetd is not installed (Automated)
+    title: Ensure autofs services are not in use (Automated)
     levels:
       - l1_server
-      - l1_workstation
-    status: automated
+      - l2_workstation
+    status: partial
```
[the remark this answers is not in the document]
Comment (vojtapolasek) [reply controls/cis_rhel8.yml:761]: OK, I can agree with that. I will fix it.

**Comment by vojtapolasek** [comment]
Hello @marcusburghardt  and thank you for the review. I rebased and made requested changes.

File: controls/cis_rhel8.yml [hunk 5]
```diff
@@ -977,7 +977,8 @@ controls:
     notes: >-
       The rule also configures correct run level to prevent unbootable system.
     rules:
-      - xwindows_remove_packages
+      - package_xorg-x11-server-common_removed
```
Comment (marcusburghardt) [inline controls/cis_rhel8.yml:980]: I think we should remove the cis@rhel7 reference from xwindows_remove_packages.

**Comment by marcusburghardt** [comment]
> Hello @marcusburghardt and thank you for the review. I rebased and made requested changes.

Great @vojtapolasek . I had just one last comment about the last changes in 2.2.20.

File: controls/cis_rhel8.yml (hunk 5 above)
Comment (vojtapolasek) [reply controls/cis_rhel8.yml:980]: I will address RHEL 7 in a separate PR. Both approaches work there, I have just tested that. Either using xwindows_remove_packages or package_xorg-x11-server-common_removed together with xwindows_runlevel_target. The latter is more aligned with the policy.

Comment (marcusburghardt) [reply controls/cis_rhel8.yml:980]: Oh, I meant cis@rhel8. Sorry. If I am not wrong, there are some packages which depends on `xorg-x11-server-utils` and this requirement were breaking valid scenarios, so the CIS policy was updated to not remove it after a discussion about this issue. Removing only the `xorg-x11-server-common` should be enough for this requirement.

Comment (vojtapolasek) [reply controls/cis_rhel8.yml:980]: fixed

Comment (marcusburghardt) [reply controls/cis_rhel8.yml:980]: Great. Thanks.

**Review by marcusburghardt — APPROVED** [review]
Great work. Thanks @vojtapolasek . We only need to wait the CI tests.

**Comment by Mab879** [comment]
@vojtapolasek please rebase

**Comment by vojtapolasek** [comment]
Rebased.

**Review by marcusburghardt — APPROVED** [review]
Thanks for the nice work @vojtapolasek .

**Pull request merged**